tests/WE2E/run_WE2E_tests.py

```diff
@@ -162,32 +155,37 @@ def run_we2e_tests(homedir, args) -> None:
         starttime_string = starttime.strftime("%Y%m%d%H%M%S")
         test_name = os.path.basename(test).split('.')[1]
         logging.debug(f"For test {test_name}, constructing config.yaml")
-        test_cfg = load_config_file(test)
-
-        if test_cfg.get('user') is None:
-            test_cfg['user'] = {}
-        test_cfg['user'].update({"MACHINE": machine})
-        test_cfg['user'].update({"ACCOUNT": args.account})
+        test_cfg = get_yaml_config(test)
+
+        test_config_updates = {
+             "user": {
+                "MACHINE": machine,
+                "ACCOUNT": args.account,
+                 },
+             "platform": {
+                 "BUILD_MOD_FN": args.modulefile,
+                 },
+             "workflow": {
+                 "COMPILER": args.compiler,
+                 "EXPT_SUBDIR": test_name
+                 },
+             }
         if run_envir:
-            test_cfg['user'].update({"RUN_ENVIR": run_envir})
-        # if platform section was not in input config, initialize as empty dict
-        if 'platform' not in test_cfg:
-            test_cfg['platform'] = dict()
-        test_cfg['platform'].update({"BUILD_MOD_FN": args.modulefile})
-        test_cfg['workflow'].update({"COMPILER": args.compiler})
+            test_config_updates['user'].update({"RUN_ENVIR": run_envir})
         if args.expt_basedir:
-            test_cfg['workflow'].update({"EXPT_BASEDIR": args.expt_basedir})
-        test_cfg['workflow'].update({"EXPT_SUBDIR": test_name})
+            test_config_updates['workflow'].update({"EXPT_BASEDIR": args.expt_basedir})
         if args.exec_subdir:
-            test_cfg['workflow'].update({"EXEC_SUBDIR": args.exec_subdir})
+            test_config_updates['workflow'].update({"EXEC_SUBDIR": args.exec_subdir})
         if args.launch == "cron":
-            test_cfg['workflow'].update({"USE_CRON_TO_RELAUNCH": True})
+            test_config_updates['workflow'].update({"USE_CRON_TO_RELAUNCH": True})
         if args.cron_relaunch_intvl_mnts:
-            test_cfg['workflow'].update({"CRON_RELAUNCH_INTVL_MNTS": args.cron_relaunch_intvl_mnts})
+            test_config_updates['workflow'].update({"CRON_RELAUNCH_INTVL_MNTS": args.cron_relaunch_intvl_mnts})
         if args.debug_tests:
-            test_cfg['workflow'].update({"DEBUG": args.debug_tests})
+            test_config_updates['workflow'].update({"DEBUG": args.debug_tests})
         if args.verbose_tests:
-            test_cfg['workflow'].update({"VERBOSE": args.verbose_tests})
+            test_config_updates['workflow'].update({"VERBOSE": args.verbose_tests})
```

Comment:
GH won't let me supply a suggestion here because the block contains deleted lines, but could this also be a bit DRYer like
```
workflow = test_config_updates['workflow']
if args.expt_basedir:
    workflow["EXPT_BASEDIR"] = args.expt_basedir
if args.exec_subdir:
    workflow["EXEC_SUBDIR"] = args.exec_subdir
```
etc. ? Assigning the sub-dict to a variable cuts down on the visual noice, and I didn't see a reason to use `.update()` instead of just `dict` key=value assignment.

Though `.update()` might be useful in an even more concise version:
```
update = lambda k, v: v and test_config_updates["workflow"].update({k: v})
update("EXPT_BASEDIR", args.expt_basedir)
update("EXEC_SUBDIR", args.exec_subdir)
```
etc. like
```
>>> args = argparse.Namespace(a=1, b=None, c=2)
>>> vars = {}
>>> f = lambda k, v: v and vars.update({k: v})
>>> f("A", args.a)
>>> f("B", args.b)
>>> f("C", args.c)
>>> vars
{'A': 1, 'C': 2}
```